app/services/hyrax/collections/migration_service.rb

```diff
@@ -74,8 +74,8 @@ def self.repair_migrated_collections
       #
       # @param collection [::Collection] collection object to be migrated/repaired
       def self.repair_migrated_collection(collection)
-        return if collection.collection_type_gid.present? && collection.collection_type_gid != Hyrax::CollectionType.find_or_create_default_collection_type.gid
-        collection.collection_type_gid = Hyrax::CollectionType.find_or_create_default_collection_type.gid
+        return if collection.collection_type_gid.present? && collection.collection_type_gid != Hyrax::CollectionType.find_or_create_default_collection_type.to_global_id
```

Comment:
Do you need to use `.to_s` on the result of `.to_global_id` because `.collection_type_gid` is a string and not a `GlobalID`?

Reply:
yeah, that's right.

Reply:
I gave my approval for this, although it looks like this change is still pending.

Reply:
I wonder if there should be a test for this.  Seems like one of the tests should have failed when collection.collection_type_gid didn't equal Hyrax::CollectionType.find_or_create_default_collection_type.to_global_id because the first was a String and the later was a GlobalID.

Reply:
Looking at this more, the line after this sets `collection.collection_type_gid` using `to_global_id` which appears to imply that the gid in the collection is a GlobalID.

Reply:
And the line 63 in the collections_controller also sets it to GlobalID...
```
@collection.collection_type_gid = CollectionType.find(collection_type_id).to_global_id
```

Is this an issue for backward compatibility where collection_type_gid is a String?

Reply:
> Is this an issue for backward compatibility where collection_type_gid is a String?

`#collection_type_gid=` casts to a string, right?

Reply:
I was concerned about this line because it looks like it is a String != GlobalID comparison.  Am I misreading this line or will a GlobalID automatically cast so it isn't an issue?

Reply:
let's just render the issue moot and revert the change in this line and the test for this service.

the whole thing is dead code that probably should have been deprecated for removal somewhere around Hyrax 2.2.0. https://github.com/samvera/hyrax/pull/4716

Reply:
Good point.  Sounds good to me.